sense-android-library/test/java/nl/sense_os/datastorageengine/CommonSenseProxyTest.java

```diff
@@ -0,0 +1,560 @@
+        thrown.expect(IOException.class);
+        thrown.expectMessage("getAllSensors: invalid input of sessionID");
+        // log in first in order to create sensor
+        proxy.loginUser(newUser.get("username"), newUser.get("password"));
+
+        // check the sensor with valid session id
+        JSONArray sensorList = proxy.getAllSensors("");
+    }
+    @Test
+    public void testAddSensorWithValidParams() throws IOException, JSONException {
+        // log in first in order to create sensor
+        String session_id = proxy.loginUser(newUser.get("username"), newUser.get("password"));
+        String name = "test";
+        String displayName = "test";
+        String deviceType = "deviceType";
+        String dataType = "json";
+        String sessionID = session_id;
+        String value = "{\"interval\":0,\"data\":[2.23, 19.45, 20.2]}";
+        String dataStructure = createDataStructure(value);
+
+        // first check the sensor JSONObject has been created,
+        JSONObject sensor = proxy.createSensor(name, displayName, deviceType, dataType, dataStructure, sessionID);
+        assertNotNull("Failed to create a sensor", sensor);
+
+        // then check if the sensorid has been returned from the server
+        String sensorId = sensor.get("sensor_id").toString();
+        assertEquals("The new sensorid returned from the server is empty",false, sensorId.isEmpty());
+
+        // add the sensor to a device
+        boolean result = proxy.addSensor(sensorId,"deviceType","uuid",sessionID);
+        assertEquals("Failed to add the sensor to a device", true,result);
+    }
+    @Test
+    public void testAddSensorWithNullUuid() throws IOException, JSONException {
+        thrown.expect(IOException.class);
+        thrown.expectMessage(startsWith("invalid input of csSensorID"));
+        // log in first in order to create sensor
+        String session_id = proxy.loginUser(newUser.get("username"), newUser.get("password"));
+        String name = "test";
+        String displayName = "test";
+        String deviceType = "deviceType";
+        String dataType = "json";
+        String sessionID = session_id;
+        String value = "{\"interval\":0,\"data\":[2.23, 19.45, 20.2]}";
+        String dataStructure = createDataStructure(value);
+
+        // first check the sensor JSONObject has been created,
+        JSONObject sensor = proxy.createSensor(name, displayName, deviceType, dataType, dataStructure, sessionID);
+        assertNotNull("Failed to create a sensor", sensor);
+
+        // then check if the sensorid has been returned from the server
+        String sensorId = sensor.get("sensor_id").toString();
+        assertEquals("The new sensorid returned from the server is empty",false, sensorId.isEmpty());
+
+        // add the sensor to a device
+        boolean result = proxy.addSensor(sensorId,"deviceType",null,sessionID);
+        assertEquals("Failed to add the sensor to a device", true,result);
```

Comment:
@excaliburnan test if it has also be added by getting devices with their sensors.